cmake/commonPresets.json

```diff
@@ -0,0 +1,76 @@
+{
+	"$schema": "https://cmake.org/cmake/help/latest/_downloads/3e2d73bff478d88a7de0de736ba5e361/schema.json",
+	"version": 8,
+	"configurePresets": [
+		{
+			"name": "base",
+			"hidden": true,
+			"binaryDir": "build",
+			"installDir": "install",
+			"cacheVariables": {
+				"Launcher_BUILD_PLATFORM": "custom"
+			}
+		},
+		{
+			"name": "base_debug",
+			"hidden": true,
+			"inherits": [
+				"base"
+			],
+			"cacheVariables": {
+				"CMAKE_BUILD_TYPE": "Debug"
```

Comment:
Can you add `CMAKE_EXPORT_COMPILE_COMMANDS=ON` and maybe `DEBUG_ADDRESS_SANITIZER=ON`?

Reply:
> Can you add CMAKE_EXPORT_COMPILE_COMMANDS=ON

Not needed after https://github.com/PrismLauncher/PrismLauncher/commit/8a21b079159e12308928dab6d6d44d89c5bac0a3

> maybe DEBUG_ADDRESS_SANITIZER=ON

CI is using this now, so no, not here. It'd need to be a separate profile thingy

Reply:
> Not needed after https://github.com/PrismLauncher/PrismLauncher/commit/8a21b079159e12308928dab6d6d44d89c5bac0a3

That is a weird condition: if there were some parentheses, it would help.

>CI is using this now, so no, not here. It'd need to be a separate profile thingy

The CI is not using the `base_debug` preset. The CI is using `base_release` .
But even then it would be nice to have a profile with that enabled for debug purposes